src/client_side.cc

```diff
@@ -1405,9 +1405,6 @@ ConnStateData::parseHttpRequest(const Http1::RequestParserPointer &hp)
         /* internal URL mode */
         /* prepend our name & port */
         http->uri = xstrdup(internalLocalUri(nullptr, hp->requestUri()));
-        // We just re-wrote the URL. Must replace the Host: header.
-        //  But have not parsed there yet!! flag for local-only handling.
-        http->flags.internal = true;
```

Comment:
I assume you have tested that the PR code does the same thing as the official code after we get here with a relative URL and the Host header that does not match the URL authority component created by internalLocalUri(). If you have not tested that, please do!

Reply:
I have tested that, yes it does AFAICT.